llmobs: clarify we only need evals tag for otel spans when submitted directly to api (#33319)
DataDog/documentation: content/en/llm_observability/evaluations/external_evaluations.md

<div class="alert alert-info">

> - Evaluation labels must be unique for a given LLM application (<code>ml_app</code>) and organization.

Reviewer: ```suggestion
Evaluation labels must be unique for a given LLM application (<code>ml_app</code>) and organization.
```

You can use the evaluations API provided by LLM Observability to send evaluations associated with spans to Datadog. See the [Evaluations API][2] for more details on the API specifications.

> <div class="alert alert-info">Submitting evaluations for <a href="/llm_observability/instrumentation/otel_instrumentation">OpenTelemetry spans</a> directly to the evaluations API requires adding the <code>source:otel</code> tag to your evaluation.</div>

Reviewer: ```suggestion
To submit evaluations for <a href="/llm_observability/instrumentation/otel_instrumentation">OpenTelemetry spans</a> directly to the Evaluations API, you must include the <code>source:otel</code> tag in the evaluation.
```
